### PR TITLE
Fix prepared statements disabled test again

### PR DESCRIPTION
**Pull request by prathamesh-sonpatki**
### Summary

- Due to `assert_nothing_raised` this test was not really testing
  anything.
- So updated it to assert that the query gives expected result.
- Also in general we can use `connection.unprepared_statement` for
  testing queries w/o prepared statements but it can't be used in this
  case. This test cases was added because when prepared_statements
  config is set to false, then DetermineIfPreparableVisitor module
  does not extended by Arel visitor resulting into an error. Ref: https://github.com/rails/rails/pull/22748.
- Because DetermineIfPreparableVisitor module does not get added to the
  visitor chain only if prepared_statements is false while **setting up
  connection**, not when `unprepared_statement` is used.
- I have also added an assertion for making sure that prepared_config
  is set to false from the start, so that nobody accidentally removes
  the connection setup using `arunit_without_prepared_statements` and
  replaces it with stubs or unprepared_statement.

### CODE REVIEW

File: activerecord/test/cases/adapters/postgresql/prepared_statements_disabled_test.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  def teardown
+    @conn.release_connection
+    ActiveRecord::Base.establish_connection :arunit
+  end
+
+  def test_select_query_works_even_when_prepared_statements_are_disabled
+    david = developers(:david)
+
+    assert_equal false, Developer.connection.prepared_statements
```
Comment (vipulnsward) [inline activerecord/test/cases/adapters/postgresql/prepared_statements_disabled_test.rb]: `refute Developer.connection.prepared_statements`

File: activerecord/test/cases/adapters/postgresql/prepared_statements_disabled_test.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    developer = Developer.where(id: 1).to_a
+    assert_equal david, developer.last
```
Comment (vipulnsward) [inline activerecord/test/cases/adapters/postgresql/prepared_statements_disabled_test.rb]: Can you add these two variants instead- https://github.com/rails/rails/pull/27079/commits/8c46f1f5e85640335f3e94ec776507e80fcc839f

with + without binds.

File: activerecord/test/cases/adapters/postgresql/prepared_statements_disabled_test.rb (hunk 1 above)
Comment (kaspth) [inline activerecord/test/cases/adapters/postgresql/prepared_statements_disabled_test.rb]: We generally avoid refute and use assert_not instead :)

**Comment by prathamesh-sonpatki** [comment]
Updated, thanks @vipulnsward

**Comment by kaspth** [comment]
I hate this test so much 😄 — thanks for all the help here, gang!

**Pull request merged**